contracts/EscrowAndStaking/CrossChainCollector.sol

```diff
@@ -0,0 +1,407 @@
+        chainRevenueInPeriod[period][_chainSelector] = _revShare;
+
+        calculatedStakedAmountForPeriod[period] += _stakedAmount;
+        calculatedEscrowedAmountForPeriod[period] += _escrowedAmount;
+        calculatedBonusPointsForPeriod[period] += _bonusPoints;
+        calculatedRevenueForPeriod[period] += _revShare;
+    }
+
+    function _broadcastMessageToAll() internal {
+        uint chainBaseRewards;
+        uint chainExtraRewards;
+        uint revShare;
+        bytes memory message;
+
+        for (uint i = 0; i < numOfActiveCollectors; i++) {
+            chainBaseRewards =
+                ((chainStakedAmountInPeriod[period][chainSelector[i]] +
+                    chainEscrowedAmountInPeriod[period][chainSelector[i]]) * baseRewardsPerPeriod) /
+                (calculatedStakedAmountForPeriod[period] + calculatedEscrowedAmountForPeriod[period]);
+
+            chainExtraRewards =
+                (chainBonusPointsInPeriod[period][chainSelector[i]] * extraRewardsPerPeriod) /
+                (calculatedBonusPointsForPeriod[period]);
+
+            revShare =
+                ((chainStakedAmountInPeriod[period][chainSelector[i]] +
+                    chainEscrowedAmountInPeriod[period][chainSelector[i]]) * calculatedRevenueForPeriod[period]) /
+                (calculatedStakedAmountForPeriod[period] + calculatedEscrowedAmountForPeriod[period]);
+
+            chainBaseRewardsInPeriod[period][chainSelector[i]] = chainBaseRewards;
+            chainExtraRewardsInPeriod[period][chainSelector[i]] = chainExtraRewards;
+            chainRevenueShareInPeriod[period][chainSelector[i]] = revShare;
+
+            message = abi.encode(chainBaseRewards, chainExtraRewards, revShare);
+            if (i == 0) {
+                _updateRewards(chainBaseRewards, chainExtraRewards, revShare);
+            } else {
+                _sendMessageToChain(chainSelector[i], message);
+            }
+        }
+    }
+
+    function _sendMessageToChain(uint64 _chainSelector, bytes memory _message) internal returns (bytes32 messageId) {
+        Client.EVM2AnyMessage memory evm2AnyMessage = Client.EVM2AnyMessage({
+            receiver: abi.encode(collectorAddress[_chainSelector]), // ABI-encoded receiver address
+            data: _message, // ABI-encoded string
+            tokenAmounts: new Client.EVMTokenAmount[](0), // Empty array indicating no tokens are being sent
+            extraArgs: Client._argsToBytes(
+                // Additional arguments, setting gas limit and non-strict sequencing mode
+                Client.EVMExtraArgsV1({gasLimit: gasLimitUsed, strict: false})
+            ),
+            // Set the feeToken  address, indicating LINK will be used for fees
+            feeToken: address(0)
+        });
+
+        // Get the fee required to send the message
+        uint256 fees = s_router.getFee(_chainSelector, evm2AnyMessage);
+
+        if (fees > address(this).balance) revert NotEnoughBalance(address(this).balance, fees);
```

Comment:
Can this be monitored, like an event if not I would maybe add an event so we can monitor, but if we know the address we can do it for an address like CL Node address?

Reply:
i guess then we can monitor that event, thanks